spec/tictactoe_spec.rb

```diff
@@ -7,7 +7,22 @@
   let (:test_ttt) { TicTacToe.new(game_io, player1, player2) }
 
   it "runs the game loop " do
-    allow(test_ttt.game_loop).to receive(:game_loop) { }
-    test_ttt.game_loop
+    expect(test_ttt.game_loop).to receive(:game_loop).once
+    expect(test_ttt).to receive(:play_again?).and_return(false)
+    test_ttt.play_game
+  end
+
+  describe "the play_again? method " do
```

Comment:
Ditto Mr. Mike again. You're already injecting the thing that decides whether or not to play again (the `game_io`) - so you should easily be able to stub that to return true the first time and false the second time, and then somehow assert that the game was played twice.

Example:
`allow(something).to receive(:some_method).and_return( 1, 2, 3 )`
will return 1 the first time `some_method` is called, 2 the second time, and then 3 for all subsequent calls to `some_method`